ably/ably_test.go

```diff
@@ -112,6 +113,85 @@ func assertNil(t *testing.T, object interface{}) {
 	}
 }
 
+// ObjectsAreEqual determines if two objects are considered equal.
+//
+// This function does no assertion of any kind.
+func objectsAreEqual(expected, actual interface{}) bool {
+
+	if expected == nil || actual == nil {
+		return expected == actual
+	}
+	if exp, ok := expected.([]byte); ok {
+		act, ok := actual.([]byte)
+		if !ok {
+			return false
+		} else if exp == nil || act == nil {
+			return exp == nil && act == nil
+		}
+		return bytes.Equal(exp, act)
+	}
+	return reflect.DeepEqual(expected, actual)
+
+}
+
+// ElementsMatch asserts that the specified listA(array, slice...) is equal to specified
+// listB(array, slice...) ignoring the order of the elements. If there are duplicate elements,
+// the number of appearances of each of them in both lists should match.
+//
+// assert.ElementsMatch(t, [1, 3, 2, 3], [1, 3, 3, 2]))
+//
+// Returns whether the assertion was successful (true) or not (false).
+func assertElementsMatch(t *testing.T, listA, listB interface{}) (ok bool) {
```

Comment:
It would be simpler to just do this:

```go
reflect.DeepEqual(sort.Strings(a), sort.Strings(b))
reflect.DeepEqual(sort.Ints(a), sort.Ints(b))
// etc.
```